h5py_wrapper/lib.py

```diff
@@ -10,25 +10,37 @@
 import tarfile
 
 
-def get_previous_version(version):
+def get_previous_version(version, path):
     """
     Retrieves the given version of the wrapper from github as a tar
     archive and extracts its contents to the current directory.
 
+    Parameters
+    ----------
+    version : str
+        Version number in format 'X.X.X'
+        Note that for version 0.0.1, the it has
+        to be specified as 'v0.0.1'.
+    path : str
+        Path to store the files.
     """
-    base_url = "https://github.com/INM-6/h5py_wrapper/archive/v"
+    base_url = "https://github.com/INM-6/h5py_wrapper/archive/"
     r = requests.get(''.join((base_url, version, ".tar.gz")))
+    # Convert path to str
+    path = str(path)
     try:
         r.raise_for_status()
-        fn = ''.join((os.path.join(os.getcwd(), version), '.tar.gz'))
+        fn = ''.join((os.path.join(path, version), '.tar.gz'))
         with open(fn, 'wb') as f:
             f.write(r.content)
         with tarfile.open(fn) as f:
-            f.extract(''.join(('h5py_wrapper-', version, '/wrapper.py')))
-            f.extract(''.join(('h5py_wrapper-', version, '/__init__.py')))
-        os.rename('-'.join(('h5py_wrapper', version)),
-                  '_'.join(('h5py_wrapper', version.replace('.', ''))))
-        os.remove(fn)
+            f.extractall(path=path)
+        # This is necessary to account for the fact that version 0.0.1 was called v0.0.1
+        # but all subsequent version are called X.X.X without the leading v.
```

Comment:
ouch, bad call from us on changing the convention. would it be possible to fix this retrospectively by modifying the release? but I guess this causes more problems than it solves.

Reply:
I don't think that it is possible. After all, it's the purpose of a release that it can't be changed, isn't it?

Reply:
true. let's not mess with that. we know better next time.